src/tree/updater_quantile_hist.cc

```diff
@@ -880,19 +877,23 @@ void QuantileHistMaker::Builder::PartitionKernel(
                                                                  range.begin(), range.end());
   const bst_uint fid = tree[nid].SplitIndex();
   const bool default_left = tree[nid].DefaultLeft();
-  const auto column = column_matrix.GetColumn(fid);
-  const uint32_t offset = column.GetBaseIdx();
-  common::Span<const uint32_t> idx_spin = column.GetFeatureBinIdxPtr();
+  const auto column = column_matrix.GetColumn<BinIdxType>(fid);
 
+  const int32_t offset = column.GetBaseIdx();
+  const std::vector<bool>* missing_val_flag = column.missing_flags_;
+  common::Span<const BinIdxType> idx_span = column.GetFeatureBinIdxPtr();
+  const size_t disp = column.missing_flags_offset_;
   std::pair<size_t, size_t> child_nodes_sizes;
 
   if (column.GetType() == xgboost::common::kDenseColumn) {
     if (default_left) {
       child_nodes_sizes = PartitionDenseKernel<true>(
-                            rid_span, idx_spin, split_cond, offset, left, right);
+                            rid_span, idx_span, split_cond, offset, left,
```

Comment:
Why not just pass the `column` into this kernel instead of getting all these parameters?

Reply:
done